notification/tests.py

```diff
@@ -15,7 +15,29 @@ def test_delete_notification_from_database(self, notification):
         with pytest.raises(Notification.DoesNotExist):
             Notification.objects.get(pk=notification.pk)
 
-    def test_mark_as_read_notification(self, notification):
+    def test_update_notification_message(self, notification):
+        new_notification = notification
+        assert notification.message == 'Test notification'
+        notification.message = 'ABCDE'
+        notification.save()
+        updated_notification = Notification.objects.get(pk=new_notification.pk)
+        assert updated_notification.message == 'ABCDE'
+        assert new_notification.id == notification.id == updated_notification.id
+
+    def test_update_notification_type(self, notification):
+        new_notification = notification
+        assert notification.notification_type == NotificationType.WEBSITE
+        notification.notification_type = NotificationType.CHAT
+        notification.save()
+        updated_notification = Notification.objects.get(pk=new_notification.pk)
+        assert updated_notification.notification_type == NotificationType.CHAT
+        assert new_notification.id == notification.id == updated_notification.id
+
+    def test_update_notification_is_read(self, notification):
+        new_notification = notification
         assert notification.is_read is False
-        notification.mark_notification_as_read()
-        assert notification.is_read is True
```

Comment:
What's the reason for changing this? What are we testing here?

Reply:
The reason for the change is that in the old version the test was pretty small.
I used the type of tests that I've used in another PRs and that have been approved before: making a new_entity variable, updating the data and then making sure it was updated successfully.

Reply:
A couple of things that caught my eye:

- you were testing `mark_notification_as_read()` before, but now you're not testing it anymore. What's the reason for that? Do we have enough test coverage for that function.
- when you assign `notification` into `new_notification`, you're making both variables contain the same pointer, so when you change something within one, you're changing the other as well, so there's no sense in asserting they are the same in the end.
This pattern was already problematic on previous PRs, but we did not catch it on time unfortunately.

Reply:
If we don't use `mark_notification_as_read()` anymore, then we should delete it. Currently we just remove its tests which means we're introducing untested code to out code base.

About the pointer, I think your best bet will be to copy the internal fields you're interested with. Do some sanity checks with things that you expect to fail so that you convince yourself that you're not comparing something to itself.

Reply:
Good catch! I haven't noticed that mark_notification_as_read() is unused. 
About the pointer - I think that I'll skip the "entity copy" for now, and just save the ID before the changes and make sure it wasn't changed during the update operation.